semgrep/semgrep/constants.py

```diff
@@ -46,17 +46,18 @@ def is_json(self) -> bool:
 NOSEM_INLINE_RE = re.compile(
     # We're looking for items that look like this:
     # ' nosem'
-    # ' nosem: example-pattern-id'
+    # ' nosemgrep: example-pattern-id'
     # ' nosem: pattern-id1,pattern-id2'
-    # ' NOSEM:pattern-id1,pattern-id2'
+    # ' NOSEMGREP:pattern-id1,pattern-id2'
     #
     # * We do not want to capture the ': ' that follows 'nosem'
     # * We do not care about the casing of 'nosem'
     # * We want a comma-separated list of ids
     # * We want multi-language support, so we cannot strictly look for
     #   Python comments that begin with '# '
+    # * nosem and nosemgrep should be interchangeable
     #
-    r" nosem(?::[\s]?(?P<ids>([^,\s](?:[,\s]+)?)+))?",
+    r" nosem(?:grep)?(?::[\s]?(?P<ids>([^,\s](?:[,\s]+)?)+))?",
```

Comment:
Could make the user experience much less (potentially) frustrating by being more forgiving of whitespace here.

Without this change then `//nosem:  some-rule-id` (two spaces there) will be treated as just `// nosem`.
```suggestion
    r" nosem(?:grep)?(?::[\s]*(?P<ids>([^,\s](?:[,\s]+)?)+))?",
```